include/gridtools/storage/sid.hpp

```diff
@@ -33,7 +33,8 @@ namespace gridtools {
 
             constexpr dimension_kind get_dimension_kind(int i, size_t n_dim) {
                 return i < 0 ? dimension_kind::masked
-                             : i + 1 == n_dim ? dimension_kind::innermost : dimension_kind::dynamic;
+                             : static_cast<std::size_t>(i) + 1 == n_dim ? dimension_kind::innermost
```

Comment:
will changing `n_dim` parameter to `int` work as well?

Reply:
done